core/api/serializers.py

```diff
@@ -0,0 +1,140 @@
+"""Serializers for the Product APIs
+
+This file contains all the serializers for the Product APIs. The
+CategorySerializer is used to create the category. The AllCategorySerializer is
+used to format the all the categories data into json format. The
+ProductSerializer is used to create the product and so on.
+"""
+
+from rest_framework.serializers import ModelSerializer
+from core.models import (
+    Cart, Category, Order, OrderItem, Product, ProductReview
+)
+
+
+class CategoryViewSerializer(ModelSerializer):
+    """Category View Serializer
+
+    This class is used to format the category data. The category data is
+    returned in a json format including the category id and name.
+    """
+
+    class Meta:
+        """Meta class for the Category View Serializer"""
+
+        model = Category
+        fields = ["id", "name"]
+
+
+class ProductViewSerializer(ModelSerializer):
+    """Product View Serializer
+
+    This class is used to format the product data. The product data is
+    returned in a json format including the product id, name, price, category,
+    stock, sold and description.
+    """
+
+    class Meta:
+        """Meta class for the ProductViewSerializer"""
+
+        model = Product
+        fields = [
+            "id",
+            "name",
+            "price",
+            "category",
+            "stock",
+            "sold",
+            "description",
+            "image",
+        ]
+
+    def to_representation(self, instance):
```

Comment:
Try using the other approach, as discussed in person, instead of overriding `to_representation`.

Something like: 

`category = serializers.CharField(source="category.name") `
or if you want all fields:
 `category = CategoryViewSerializer(read_only=True)`

Also look into `serializers.SerializerMethodField`